src/Raven.Server/Documents/DocumentsTransaction.cs

```diff
@@ -47,17 +47,22 @@ public DocumentsTransaction(DocumentsOperationContext context, Transaction trans
         }
 
         public override void BeforeCommit()
+        {
+            if (_attachmentHashesToMaybeDelete == null)
+                return;
+
+            _context.DocumentDatabase.DocumentsStorage.AttachmentsStorage.RemoveAttachmentStreamsWithoutReferences(_context, _attachmentHashesToMaybeDelete);
+        }
+
+        protected override void AfterCommit()
         {
             if (_executeDocumentsMigrationBeforeCommit)
```

Comment:
so it is before or after?

Reply:
sorry, fixed on https://github.com/ravendb/ravendb/pull/17210